File: Practice/Scala/src/CodingChallenges/DrawTriangles.scala
package io.beansprout
package CodingChallenges

@main
def main(): Unit = {
  val triangleFactory = new DrawTriangles
  triangleFactory.makeTriangles(6)
}

class DrawTriangles {
  def makeTriangles(numOfTriangles: Int): Unit = {
    val aster = "*"

    for (_ <- 1 to 3) { //starts next triangle
      var t = 1

      for (_ <- 1 to numOfTriangles) { //creates a new line
        for (_ <- 1 to t) { //loops asterisks
          print(aster)
        }

        println()
        t += 2
      }
    }
  }
}

File: Practice/Java/src/CodingChallenges/DrawTriangles.java
package io.beansprout.CodingChallenges;

public class DrawTriangles {
    public static void main(String[] args) {
        DrawTriangles triangleFactory = new DrawTriangles();

        triangleFactory.makeTriangles(4);
    }

    public void makeTriangles(int rows) {
        String aster = ("*");

        for (int tri = 1; tri <= 3; tri++) {//starts next triangle
            int t = 1;

            for (int s = 1; s <= rows; s++) { //creates a new line

                for (int l = 1; l <= t; l++) { //loops asterisks
                    System.out.print(aster);
                }

                System.out.println();
                t += 2;
            }
        }
    }
}
